gauntlet/packages/gauntlet-solana/src/commands/internal/solana.ts

```diff
@@ -86,24 +139,204 @@ export default abstract class SolanaCommand extends WriteCommand<TransactionResp
       price?: number
     } = {},
   ): Promise<TransactionSignature> => {
-    const { blockhash, lastValidBlockHeight } = await this.provider.connection.getLatestBlockhash()
     if (overrides.units) logger.info(`Sending transaction with custom unit limit: ${overrides.units}`)
     if (overrides.price) logger.info(`Sending transaction with custom unit price: ${overrides.price}`)
+
+    this.simulateTx(this.wallet.publicKey, rawTxs)
+
+    const currentBlockhash = await this.provider.connection.getLatestBlockhash()
+
     const tx = makeTx(
-      rawTxs,
       {
-        blockhash,
-        lastValidBlockHeight,
-        feePayer: this.wallet.publicKey,
+        instructions: rawTxs,
+        recentBlockhash: currentBlockhash.blockhash,
+        payerKey: this.wallet.publicKey,
       },
       overrides,
     )
     if (extraSigners) {
-      tx.sign(...extraSigners)
+      tx.sign(extraSigners)
     }
-    const signedTx = await this.wallet.signTransaction(tx)
+    const signedTx = await this.wallet.signVersionedTransaction(tx)
     logger.loading('Sending tx...')
-    return await sendAndConfirmRawTransaction(this.provider.connection, signedTx.serialize())
+
+    const rawTransaction = signedTx.serialize()
+
+    const txid = await this.provider.connection.sendRawTransaction(rawTransaction, {
+      skipPreflight: true,
+    })
+
+    console.log('Started awaiting confirmation for', txid, 'size:', rawTransaction.length)
+    const startTime = getUnixTs()
+    let timeout = 60_000
+    let done = false
+    let retryAttempts = 0
+    const retrySleep = 2000
+    const maxRetries = 30
+    ;(async () => {
+      while (!done && getUnixTs() - startTime < timeout / 1000) {
+        await sleep(retrySleep)
+        // console.log(new Date().toUTCString(), ' sending tx ', txid);
+        this.provider.connection.sendRawTransaction(rawTransaction, {
+          skipPreflight: true,
+        })
+        if (retryAttempts <= maxRetries) {
+          retryAttempts = retryAttempts++
+        } else {
+          break
+        }
+      }
+    })()
```

Comment:
this could be perfectly abstracted into a `sendWithRetries` method:
```suggestion
    let done = false
    const isDone = () => done
    const timeout = 60_000
    const txid = await this.sendWithRetries(isDone, rawTransaction, timeout)
```